src/allocator.rs

```diff
@@ -611,13 +629,19 @@ impl Allocator {
             match node.object_type() {
                 ObjectType::Pair => {
                     self.u8_vec.truncate(start);
-                    return err(*node, "(internal error) concat expected atom, got pair");
+                    return Err(EvalErr::InternalError(
+                        *node,
+                        "concat expected atom, got pair".to_string(),
+                    ))?;
                 }
                 ObjectType::Bytes => {
                     let term = self.atom_vec[node.index() as usize];
                     if counter + term.len() > new_size {
                         self.u8_vec.truncate(start);
-                        return err(*node, "(internal error) concat passed invalid new_size");
+                        return Err(EvalErr::InternalError(
+                            *node,
+                            "concat passed invalid new_size".to_string(),
```

Comment:
if we stick with `String` in the errors, we could include more information here, like `counter` and `term.len()`